modules/core/src/norm.cpp

```diff
@@ -630,7 +631,7 @@ double cv::norm( InputArray _src, int normType, InputArray _mask )
     normType &= NORM_TYPE_MASK;
     CV_Assert( normType == NORM_INF || normType == NORM_L1 ||
                normType == NORM_L2 || normType == NORM_L2SQR ||
-               ((normType == NORM_HAMMING || normType == NORM_HAMMING2) && _src.type() == CV_8U) );
+               ((normType == NORM_HAMMING || normType == NORM_HAMMING2) && _src.type() == CV_8UC1) );
```

Comment:
Looks like bug is catched here from the original code.

Why not `_src.depth() == CV_8U` ?

/cc @vpisarev

Reply:
You will see hundreds of similar cases! There are even functions that receive type, although they just need the depth... I have actually made several decisions in utilizing depth instead of type (e.g. in all arithmetic operations, `convertTo`, `assign`, `assignTo`, `calcCovarMatrix`), and I really hop you will catch what I missed.